### ISSUE TITLE
Caching not working with static::render in routes definition

### ISSUE THREAD

**Issue by isjackwild** [getkirby/kirby#496]
I'm using a static render method in a route definition to render some JSON from a page template and pass it into the home template.

```
$data = array();
$data['initData'] = array(
        'collection' => static::render($collection), // Render the start collection template JSON (site/templates/tour-dates.php)
        'dates' => static::render($dates), // Render the Tour Dates template JSON (site/templates/tour-dates.php)
        'slug' => $slug,
      );

return array('home', $data); // return the home page, passing in the dates and collection and initial slug data (site/templates/home.php);
```

This is working perfectly — when I visit the site, the homepage template is rendered, and the data is passed into the template.

However when I turn on caching (`c::set('cache', true);`), the collection page is rendered and returned and cached, rather than the homepage. (It will always return at the first static::render() it finds — so if I move dates above collections, it will return the dates page JSON.

**Comment by lukasbestle** [comment]
Could you please post a bit more context of and for your code? I don't really understand what you are trying to do?

**Comment by lukasbestle** [comment]
Closing this as there is a topic in the forum about this. Please don't post the same issue twice.

**Comment by isjackwild** [comment]
Sorry. :)


On 21 Sep 2016, at 18:33, Lukas Bestle <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #496<https://github.com/getkirby/kirby/issues/496>.
